### PR TITLE
Dto's, controller, mapper, exception were added

### PR DESCRIPTION
**Pull request by slizko1**

### CODE REVIEW

File: src/main/java/com/samoilenko/onlinebookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.samoilenko.onlinebookstore.dto;
+
+import java.math.BigDecimal;
+
+public class CreateBookRequestDto {
+    private Long id;
```
Comment (makaroshyna) [inline src/main/java/com/samoilenko/onlinebookstore/dto/CreateBookRequestDto.java]: Request DTO should be without id. I don't think that assigning ids when creating a book is a good idea.

**Review by makaroshyna — APPROVED** [review]
Really good! Just remove ID from request DTO :)

File: src/main/java/com/samoilenko/onlinebookstore/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+public class BookController {
+
+    private final BookService bookService;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/books/{id}")
```
Comment (HAIOVYI) [inline src/main/java/com/samoilenko/onlinebookstore/controller/BookController.java]: it look likes a dublicate `/books/`

it's already defined in the class lvl

File: src/main/java/com/samoilenko/onlinebookstore/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (HAIOVYI) [reply src/main/java/com/samoilenko/onlinebookstore/dto/CreateBookRequestDto.java]: agree

File: src/main/java/com/samoilenko/onlinebookstore/exception/EntityNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.samoilenko.onlinebookstore.exception;
+
+public class EntityNotFoundException extends RuntimeException {
+    public EntityNotFoundException(String message) {
+        super(message);
+    }
+}
```
Comment (HAIOVYI) [inline src/main/java/com/samoilenko/onlinebookstore/exception/EntityNotFoundException.java]: maybe add `public EntityNotFoundException(String message, Throwable cause) {
        super(message, cause);
    }`
    and use `Throwable cause`

**Review by HAIOVYI — CHANGES_REQUESTED** [review]
GJ!

File: src/main/java/com/samoilenko/onlinebookstore/controller/BookController.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping("/books")
```
Comment (andrii-hoienko) [inline src/main/java/com/samoilenko/onlinebookstore/controller/BookController.java]:
```suggestion
@RequestMapping("api/books")
```

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request merged**